src/Mappings/Readers/ColumnNameMatchingValueReader.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace ExcelMapper.Mappings.Readers
+{
+    using System;
+
+    using ExcelDataReader;
+
+    /// <summary>
+    /// Reads the value of a single cell given the predicate matching the column name.
+    /// </summary>
+    public sealed class ColumnNameMatchingValueReader : ICellValueReader
+    {
+        private readonly Func<string, bool> predicate;
+
+        /// <summary>
+        /// Constructs a reader that reads the value of a single cell given the predicate matching the column name.
+        /// </summary>
+        /// <param name="predicate">The predicate containing the column name to read.</param>
+        public ColumnNameMatchingValueReader(Func<string, bool> predicate)
+        {
+            if (predicate == null)
+            {
+                throw new ArgumentNullException(nameof(predicate));
+            }
+
+            this.predicate = predicate;
```

Comment:
Nit: please remove unnecessary this

Reply:
this is needed here because the field is the same name as the passed in parameter. I can rename the instance field to _predicate if you wish?